tests/unit/test_functools.py

```diff
@@ -0,0 +1,133 @@
+import pytest
+
+from common.entities.schemas import StoryItem
+from tests.test_helpers import get_test_data_from_json_file
+from utilities.exceptions import ParseGraphError
+from utilities.functools import (
+    parse_graph,
+    parse_graph_as_incidence_matrix,
+    parse_story_structure,
+)
+
+
+def test_parse_story_structure(test_data):
+    assert StoryItem(**test_data) == parse_story_structure(test_data)
+
+
+def test_parse_story_structure_with_invalid_data():
+    with pytest.raises(ParseGraphError):
+        parse_story_structure(
+            get_test_data_from_json_file(
+                "tests/unit/test_json/story_without_node_text.json"
+            )
+        )
+
+    with pytest.raises(ParseGraphError):
+        parse_story_structure(
+            get_test_data_from_json_file(
+                "tests/unit/test_json/story_without_nodes.json"
+            )
+        )
```

Comment:
1) rename to `tests/unit/test_parse_story_structure.py`
2) use pytest.mark.parametrize